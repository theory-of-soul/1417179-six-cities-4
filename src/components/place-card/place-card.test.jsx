import React from "react";
import renderer from 'react-test-renderer';
import PlaceCard from "./place-card";

const placeList = [{
  mark: `premium`,
  img: `img/apartment-01.jpg`,
  value: 120,
  time: `night`,
  isInBookmark: true,
  rating: 5,
  name: `luxurious apartment at great location`,
  type: `Apartment`
}, {
  img: `img/apartment-02.jpg`,
  value: 200,
  time: `week`,
  isInBookmark: false,
  rating: 0,
  name: `Beautiful & luxurious`,
  type: `Hotel`
}];

describe(`PlaceCard component snapshot tests`, () => {
  it(`PlaceCard component show place with a mark and in bookmark`, () => {
    const tree = renderer
      .create(<PlaceCard place={placeList[0]} onHoverHandler={() => {}}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`PlaceCard component show place without mark and not in bookmark`, () => {
    const tree = renderer
      .create(<PlaceCard place={placeList[1]} onHoverHandler={() => {}}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
