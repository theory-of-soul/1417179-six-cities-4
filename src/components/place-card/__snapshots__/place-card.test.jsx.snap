// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard component snapshot tests PlaceCard component show place with a mark and in bookmark 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="luxurious apartment at great location"
        className="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          120
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button place-card__bookmark-button--active"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
      >
        luxurious apartment at great location
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Apartment
    </p>
  </div>
</article>
`;

exports[`PlaceCard component snapshot tests PlaceCard component show place without mark and not in bookmark 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Beautiful & luxurious"
        className="place-card__image"
        height="200"
        src="img/apartment-02.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          200
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          week
        </span>
      </div>
      <button
        className="place-card__bookmark-button button "
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
      >
        Beautiful & luxurious
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Hotel
    </p>
  </div>
</article>
`;
